ds/CppBenchmarkTarget/CppBenchmarkTarget.cpp

```diff
@@ -0,0 +1,877 @@
+	/*----- PROTECTED REGION ID(CppBenchmarkTarget::read_ScalarWritesCount) ENABLED START -----*/
+	//	Set the attribute value
+	*attr_ScalarWritesCount_read = scalar_writes_count;
+	attr.set_value(attr_ScalarWritesCount_read);
+
+	/*----- PROTECTED REGION END -----*/	//	CppBenchmarkTarget::read_ScalarWritesCount
+}
+//--------------------------------------------------------
+/**
+ *	Read attribute SpectrumWritesCount related method
+ *	Description: spectrum writes count
+ *
+ *	Data type:	Tango::DevLong
+ *	Attr type:	Scalar
+ */
+//--------------------------------------------------------
+void CppBenchmarkTarget::read_SpectrumWritesCount(Tango::Attribute &attr)
+{
+	DEBUG_STREAM << "CppBenchmarkTarget::read_SpectrumWritesCount(Tango::Attribute &attr) entering... " << endl;
+	/*----- PROTECTED REGION ID(CppBenchmarkTarget::read_SpectrumWritesCount) ENABLED START -----*/
+	//	Set the attribute value
+	*attr_SpectrumWritesCount_read = spectrum_writes_count;
+	attr.set_value(attr_SpectrumWritesCount_read);
+
+	/*----- PROTECTED REGION END -----*/	//	CppBenchmarkTarget::read_SpectrumWritesCount
+}
+//--------------------------------------------------------
+/**
+ *	Read attribute ImageWritesCount related method
+ *	Description: image writes count
+ *
+ *	Data type:	Tango::DevLong
+ *	Attr type:	Scalar
+ */
+//--------------------------------------------------------
+void CppBenchmarkTarget::read_ImageWritesCount(Tango::Attribute &attr)
+{
+	DEBUG_STREAM << "CppBenchmarkTarget::read_ImageWritesCount(Tango::Attribute &attr) entering... " << endl;
+	/*----- PROTECTED REGION ID(CppBenchmarkTarget::read_ImageWritesCount) ENABLED START -----*/
+	//	Set the attribute value
+	*attr_ImageWritesCount_read = image_writes_count;
+	attr.set_value(attr_ImageWritesCount_read);
+
+	/*----- PROTECTED REGION END -----*/	//	CppBenchmarkTarget::read_ImageWritesCount
+}
+//--------------------------------------------------------
+/**
+ *	Read attribute CommandCallsCount related method
+ *	Description: command calls count
+ *
+ *	Data type:	Tango::DevLong
+ *	Attr type:	Scalar
+ */
+//--------------------------------------------------------
+void CppBenchmarkTarget::read_CommandCallsCount(Tango::Attribute &attr)
+{
+	DEBUG_STREAM << "CppBenchmarkTarget::read_CommandCallsCount(Tango::Attribute &attr) entering... " << endl;
+	/*----- PROTECTED REGION ID(CppBenchmarkTarget::read_CommandCallsCount) ENABLED START -----*/
+	//	Set the attribute value
+	*attr_CommandCallsCount_read = command_calls_count;
+	attr.set_value(attr_CommandCallsCount_read);
+
+	/*----- PROTECTED REGION END -----*/	//	CppBenchmarkTarget::read_CommandCallsCount
+}
+//--------------------------------------------------------
+/**
+ *	Read attribute TimeSinceReset related method
+ *	Description: time since reset
+ *
+ *	Data type:	Tango::DevDouble
+ *	Attr type:	Scalar
+ */
+//--------------------------------------------------------
+void CppBenchmarkTarget::read_TimeSinceReset(Tango::Attribute &attr)
```

Comment:
I have detected an issue with this attribute - after trying to read it the device stop to respond, however the server is still on-line... not yet found the reason.

Reply:
This could be some coincidence with reading image attribute according to local manual tests... Need to check it more carefully...